client/my-sites/plugins/controller-logged-out.js

```diff
@@ -36,11 +41,44 @@ const prefetchPopularPlugins = ( queryClient, options ) =>
 const prefetchFeaturedPlugins = ( queryClient ) =>
 	prefetchPluginsData( queryClient, getWPCOMFeaturedPluginsQueryParams() );
 
-const prefetchProductList = ( store ) => {
-	const productsList = getProductsList( store.getState() );
-	if ( Object.values( productsList ).length === 0 ) {
-		return requestProductsList( { type: 'all' } )( store.dispatch );
-	}
+const prefetchProductList = ( queryClient, store ) => {
+	const query = { type: 'all' };
+
+	return queryClient
+		.fetchQuery( 'products-list', () => wpcom.req.get( '/products', query ) )
+		.then(
+			( productsList ) => {
+				return store.dispatch( receiveProductsList( productsList, query.type ) );
+			},
+			() => {}
```

Comment:
In case when the fetch fails, rather that timing out, should be also skip SSR by setting `context.serverSideRender = false`, and send a 50x response to bots? In both cases (fail and timeout) the SSR-ed markup will be incomplete, because the product data are missing.

Reply:
That makes sense. We should probably do the same for the other queries.
Do you think it's worth logging the fetch fails as well?

Reply:
> About the useragent.isBot condition: maybe we want to make the timeout a bit longer for bots?

Sounds good, I set a 10s timeout for them.

Also enabled logging for the other fetch fails.